lib/actions.ts

```diff
@@ -286,17 +287,26 @@ export async function getJobObject(octokit: InstanceType<typeof GitHub>): Promis
     throw error
   })
 
-  const currentJob = workflowRunJobs.find((job) => job.name === absoluteJobName)
+  
+  const runnerName = getInput('runner-name', {required: true})
+  //In the case of truncated job name, the runner name was the only other way i could find to identify the job
+  // This still might produce a run that points to the wrong job, but it's the best I could do
+  const currentJob = workflowRunJobs.filter((job) => job.name === absoluteJobName && job.status=== "in_progress" && job.runner_name === runnerName)
   if (!currentJob) {
```

Comment:
This is always false because currentJob is an array

Reply:
Im still new to JavaScript/TypeScript, is there a better way other than checking the length to do this?